grails-app/domain/org/pih/warehouse/order/OrderAdjustment.groovy

```diff
@@ -108,8 +108,23 @@ class OrderAdjustment implements Serializable, Comparable<OrderAdjustment> {
         return invoiceItems*.invoice.unique()
     }
 
+    /**
+     * Adjustment can be on a regular invoice if it is not canceled or if it is canceled, but has prepayment
+     * */
+    Boolean canBeOnRegularInvoice() {
+        if (canceled) {
+            return hasPrepaymentInvoice
+        }
+
+        return true
```

Comment:
can be simplified to `return !canceled || hasPrepaymentInvoice` if you want

Reply:
Yeah, when I was doing the improvements to fix some edge cases, I wanted to keep it as clear as possible, to avoid any further confusion. I want to add a separate subtask for adding more tests (and improving currently existing ones) that will be done after the solution is working properly, then this could be changed to a one-liner there, if this is preferable.